cmd/main.go

```diff
@@ -399,54 +403,67 @@ func RunRootCommand(ctx context.Context, opt Options, args []string) error {
 	defer llmClient.Close()
 
 	// Initialize session management
-	var chatStore api.ChatMessageStore
+	var session *api.Session
 	var sessionManager *sessions.SessionManager
 
-	// TODO: Remove this when session persistence is default
 	if opt.NewSession || opt.ResumeSession != "" {
-		sessionManager, err = sessions.NewSessionManager()
+		sessionManager, err = sessions.NewSessionManager(opt.SessionBackend)
 		if err != nil {
 			return fmt.Errorf("failed to create session manager: %w", err)
 		}
 
-		// Handle session creation or loading
+		// Case 1: Create a completely new session
```

Comment:
This seems to be added by AI. We should avoid having "How" style comments in the code. Comments should mostly focus on the "Why" aspect here.